src/main/java/com/epam/reportportal/service/LockFile.java

```diff
@@ -0,0 +1,303 @@
+package com.epam.reportportal.service;
+
+import com.epam.reportportal.exception.InternalReportPortalClientException;
+import com.epam.reportportal.listeners.ListenerParameters;
+import com.epam.reportportal.utils.Waiter;
+import com.epam.reportportal.utils.properties.ListenerProperty;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A service to perform blocking I/O operations on '.lock' and '.sync' file to get single launch UUID for multiple clients on a machine.
+ * This class uses a local storage disk, therefore applicable in scope of a single hardware machine. You can control lock and sync file
+ * paths with {@link ListenerProperty#LOCK_FILE_NAME} and {@link ListenerProperty#SYNC_FILE_NAME} properties.
+ */
+public class LockFile implements LaunchIdLock {
+	private static final Logger LOGGER = LoggerFactory.getLogger(LockFile.class);
+
+	public static final String LOCK_FILE_CHARSET = "US-ASCII";
+	private static final String LINE_SEPARATOR = System.getProperty("line.separator");
+	private static final float MAX_WAIT_TIME_DISCREPANCY = 0.1f;
+
+	private final File lockFile;
+	private final File syncFile;
+	private final long fileWaitTimeout;
+	private volatile String lockUuid;
+	private volatile Pair<RandomAccessFile, FileLock> mainLock;
+
+	public LockFile(ListenerParameters parameters) {
+		lockFile = new File(parameters.getLockFileName());
+		syncFile = new File(parameters.getSyncFileName());
+		fileWaitTimeout = parameters.getFileWaitTimeout();
+	}
+
+	private Pair<RandomAccessFile, FileLock> obtainLock(final File file) {
+		final String filePath = file.getPath();
+		RandomAccessFile lockAccess;
+		try {
+			lockAccess = new RandomAccessFile(file, "rwd");
+		} catch (FileNotFoundException e) {
+			LOGGER.info("Unable to open '{}' file: {}", filePath, e.getLocalizedMessage(), e);
```

Comment:
Probably log level should be 'warn' or 'error'

Reply:
Not really, when a file locked for writing on Windows another process can't even open it and throws that exception. So it's a rather common case for the service. And because of that I decided to even lower the level to DEBUG :)